### PR TITLE
mv: implement move across file systems

### PR DESCRIPTION
**Pull request by rminnich**
Includes a test which can only run as root on Unix.

We might consider moving mv into the cp package at some point.

### CODE REVIEW

**Comment by rminnich** [comment]
needs 1.13

File: cmds/core/mv/mv.go [hunk 1]
```diff
@@ -59,25 +61,31 @@ func moveFile(source string, dest string) error {
 			return nil
 		}
 	}
-
 	if err := os.Rename(source, dest); err != nil {
-		return err
+		// Check for EXDEV; if so, we'll do a CopyTree
+		uerr, ok := err.(*os.LinkError)
+		if !ok || uerr.Unwrap() != unix.EXDEV {
```
Comment (hugelgupf) [inline cmds/core/mv/mv.go:67]: I would use errors.Is(err, unix.EXDEV), if possible? No need to cast, I *think* it'll go unwrap all that stuff by itself.

```go
if !errors.Is(err, unix.EXDEV) {
  return err
} 
if err := cp....
```

File: cmds/core/mv/mv_test.go [hunk 2]
```diff
@@ -281,6 +282,42 @@ func TestMvNoClobber(t *testing.T) {
 	}
 }
 
+func TestCrossFileSystems(t *testing.T) {
+	// The namespace creator does not have a teardown capability and it looked a bit
+	// tricky to add one.
+	if os.Getuid() != 0 {
```
Comment (hugelgupf) [inline cmds/core/mv/mv_test.go:288]: testutil.SkipIfNotRoot(t)

pkg/testutil

but I'd say it's optional, just a nit

File: cmds/core/mv/mv_test.go [hunk 3]
```diff
@@ -281,6 +282,42 @@ func TestMvNoClobber(t *testing.T) {
...
+	}
+	defer os.RemoveAll(d1)
+	m1, err := mount.Mount("tmpfs", d1, "tmpfs", "", 0)
+	if err != nil {
+		t.Fatalf("mount tmpfs on d1: got %v, want nil", err)
+	}
+	defer m1.Unmount(0)
+	d2, err := ioutil.TempDir("", "mvto")
+	if err != nil {
+		t.Fatalf("from dir: got %v, want nil", err)
```
Comment (hugelgupf) [inline cmds/core/mv/mv_test.go:303]: s/from dir/to dir

File: cmds/core/mv/mv_test.go [hunk 4]
```diff
@@ -281,6 +282,42 @@ func TestMvNoClobber(t *testing.T) {
...
+	}
+	defer os.RemoveAll(d2)
+	srcDir := filepath.Join(d1, "dir")
+	if err := os.Mkdir(srcDir, 0755); err != nil {
+		t.Fatalf("%s: got %v, want nil", srcDir, err)
+	}
+	srcFile := filepath.Join(srcDir, "file")
+	f, err := os.Create(srcFile)
```
Comment (hugelgupf) [inline cmds/core/mv/mv_test.go:311]: use ioutil.WriteFile to put some content in it?

File: cmds/core/mv/mv_test.go [hunk 5]
```diff
@@ -281,6 +282,42 @@ func TestMvNoClobber(t *testing.T) {
...
+	if err != nil {
+		t.Fatalf("Create %s: got %v, want nil", srcFile, err)
+	}
+	f.Close()
+	if err := mv([]string{srcFile, d2}, true, false, false); err != nil {
+		t.Fatalf("%v -> %v: got %v, want nil", srcFile, d2, err)
+	}
```
Comment (hugelgupf) [inline cmds/core/mv/mv_test.go:318]: use pkg/cp/cmp to make sure the content is the same after the move? https://godoc.org/github.com/u-root/u-root/pkg/cp/cmp#IsEqualTree

File: cmds/core/mv/mv.go (hunk 1 above)
Comment (rminnich) [reply cmds/core/mv/mv.go:67]: I'll give that a try, thanks!

**Comment by rminnich** [comment]
though I'd closed this thing, try again.

**Pull request closed without merging**